AirCasting/CoreData/MeasurementStreamStorage.swift

```diff
@@ -67,6 +91,29 @@ final class HiddenCoreDataMeasurementStreamStorage: MeasurementStreamStorageCont
         self.context = context
     }
     
+    func deleteSession(_ sessionUUID: SessionUUID) throws {
+        do {
+            try context.delete(context.existingSession(uuid: sessionUUID))
+        } catch {
+            Log.info("## Error when deleting session")
+        }
+    }
+    
+    func deleteStreams(_ sessionUUID: SessionUUID) throws {
+        let sessionEntity = try context.existingSession(uuid: sessionUUID)
+        let toDelete = sessionEntity.allStreams!.filter({ $0.gotDeleted })
+        toDelete.forEach { object in
+            do {
+                toDelete.forEach { object in
+                    context.delete(object)
+                }
+            } catch {
+                Log.info("## Error when deleting session")
+            }
+        }
+        sessionEntity.changesCount += 1
```

Comment:
Not a stupid question at all @smialko . We should actually put a comment here @jankrzempek . It's workaround for an obj-c related behavior with `NSOrderedSet`. We basically force core data to send change notifications for this Session objects in the app because the `NSOrderedSet` operations don't trigger KVO and thus don't trigger `ObservableObject` changes 🤷‍♂️ so this is the simplest workaround. Not terrible, not great

Reply:
For sure. I will allow myself to copy part of your comment @pawelgil as long as it seems great explanation.